src/charm.py

```diff
@@ -1319,10 +1319,32 @@ def set_scaling_down(self, event: RelationDepartedEvent) -> bool:
         # check if relation departed is due to current unit being removed. (i.e. scaling down the
         # application.)
         rel_departed_key = self._generate_relation_departed_key(event.relation.id)
-        scaling_down = json.dumps(event.departing_unit == self.unit)
-        self.unit_peer_data[rel_departed_key] = scaling_down
+        scaling_down = event.departing_unit == self.unit
+        self.unit_peer_data[rel_departed_key] = json.dumps(scaling_down)
         return scaling_down
 
+    def proceed_on_broken_event(self, event) -> int:
```

Comment:
The name of this method is not very intuitive imo. It's quite verbose, but what about something along the lines of: 
`def broken_relation_id_if_departed(self, event: ..) -> Optional[int]`

Reply:
You have some points here. I think given the way that it is used in other areas of the code it should be kept as is, i.e. it is used in other parts of the code as:

```
    if not self.charm.proceed_on_broken_event(event):
            logger.info("Skipping relation broken event, broken event due to scale down")
            return
```

But I agree with what you are saying here. Instead I could make this function return a `boolean` instead and process the relation id in a separte function (or even in the calling function.) What do you think?

Reply:
That sounds fair - this can be part of a next PR